src/k2/module/k23si/Module.h

```diff
@@ -171,6 +179,32 @@ class K23SIPartitionModule {
         return dto::K23SIStatus::OK("");
     }
 
+    Status _validateQueryRequest(const dto::K23SIQueryRequest& request) const {
```

Comment:
You should be able to refactor this with validateReadRequest to reduce some duplication, since only the last to checks here are specific to queries.